### ISSUE TITLE
Unable to scroll block style switcher

### ISSUE THREAD

**Issue by timelsass** [WordPress/gutenberg#16028]
When adding several variations to block styles for a single block, if the list goes beyond the visual area seen, there's no way possible for a user to see or access the rest when clicking on the block style switcher by scrolling in that area. 
![image](https://user-images.githubusercontent.com/11907254/59080385-7501fa00-88b6-11e9-8657-3e5a62cf34c0.png)

possibly related to https://github.com/WordPress/gutenberg/issues/10905

**Comment by senadir** [comment]
problem seems to be easily fixable by adding a  `max-height` and `overflow-y: auto` to `.components-popover__content` but it seems to be purposely made to `overflow-y: visible`
https://github.com/WordPress/gutenberg/blob/92fcaa82ba9a6d4663fabe98565a0eba4f4bcfc8/packages/block-editor/src/components/block-switcher/style.scss#L128-L130

this change was made in [this commit](https://github.com/WordPress/gutenberg/pull/8962/commits/49abee2e5ee7c6a21a96098d04653bcacfed1fad) to fix this issue #8623 in this pull request #8962
basically it fixes a bug in IE11

the use of `overflow-y: initial` or `overflow-y: visible` is the same, since the initial value _is_ visible, so we probably need to understand why initial was used to begin with

**Comment by senadir** [comment]
as far as I managed to dig, this fix was because the preview panel will not show up if `overflow-y` is anything other then visible.
that's because, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-x)

> If `overflow-y` is `hidden`, `scroll` or `auto` and this property is `visible` (default) it will implicitly compute to `auto`.

and if `.components-popover__content` is not visible, the preview won't show up, in this image, it is `auto`
![image](https://user-images.githubusercontent.com/6165348/59125466-55bca880-895a-11e9-8309-34d9826a0cc5.png)

so a solution I managed to make was to wrap `.components-popover__content` in a div, take `.ditor-block-preview` out of `.components-popover__content` and into the parent div, migrate some styles from the `.components-popover__content` to the parent (like background, size, position and what about)
double the width of the parent, and give `.components-popover__content` 50% of it, preserving the same visual structure but fixing it.

- This is an example, we have a popover component that is too long, I applied max height to but the overflow is still a problem
![image](https://user-images.githubusercontent.com/6165348/59125740-278b9880-895b-11e9-8c57-4e411145762c.png)

- The same example with long content, with overflow fixed and preview working
![image](https://user-images.githubusercontent.com/6165348/59126240-7b4ab180-895c-11e9-8b2c-f66a5b5d37e0.png)

this implementation still requires a lot of refactoring (since I switched from `position: absolute` to flexbox)
I will try to refactor it and make a PR
